src/megatron/energon/flavors/webdataset/joined_webdataset.py

```diff
@@ -87,27 +93,30 @@ def __init__(
             max_samples_per_sequence: Maximum number of samples per sequence (=how many samples
                     will be sequentially iterated).
             part_filter: (internal) Function for filtering tar files by dict keys
-            join_method: How to join the samples.
+            join_method: How to join the samples from the datasets.
                 inner_match: All samples must match 1:1 of the merged datasets.
                 This might be extended to further modes in the future, but those will require a new index, which
-            join_type: Type of the joined samples.
+            joiner: Type of the joined samples.
```

Comment:
Also here it says "type", I'm confused. I thought we can specify a method also.

Reply:
We can, `joiner` takes both now. Either a `Type[Sample]` or a callable.